src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,84 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            Files.deleteIfExists(Path.of(toFileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot delete file " + toFileName, e);
+        }
+
+        String[] actionsList = readFile(fromFileName);
+        String report = createReport(actionsList);
+        writeReportToFile(report, toFileName);
+    }
+    
+    private String[] readFile(String fromFileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String line = bufferedReader.readLine();
 
+            while (line != null) {
+                stringBuilder.append(line).append(System.lineSeparator());
+                line = bufferedReader.readLine();
+            }
+
+            return stringBuilder.toString().split("[\\n\\r\\t]+");
```

Comment:
Instead of using `split` with a regex, you can read lines directly in the while loop and process them, which is more efficient and cleaner. If you still need to use `split`, consider using `split(System.lineSeparator())` for splitting.